core/dbt/graph/selector.py

```diff
@@ -329,16 +338,41 @@ def select_childrens_parents(self, selected: Set[str]) -> Set[str]:
         ancestors_for = self.select_children(selected) | selected
         return self.select_parents(ancestors_for) | ancestors_for
 
-    def select_children(self, selected: Set[str]) -> Set[str]:
+    def descendants(self, node, max_depth: int = None):
+        """Returns all nodes reachable from `node` in `graph`"""
+        if not self.graph.has_node(node):
+            raise nx.NetworkXError(f"The node {node} is not in the graph.")
```

Comment:
This should raise a dbt exception, probably a `dbt.exceptions.InternalException`. That's what we generally raise when a code path shouldn't be reachable.